nvflare/app_opt/xgboost/histogram_based/executor.py

```diff
@@ -236,6 +246,10 @@ def train(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -
         self.rank = rank_map[client_name]
         self.world_size = world_size
 
+        if self.use_gpus:
+            self.log_info(fl_ctx, f"Training with GPU {self.rank}")
+            self.xgb_params["device"] = f"cuda:{self.rank}"
```

Comment:
Let's add either a comment here or in the method docstring, that we are mapping each rank to a GPU.

As enhancement later (not sure if we need), we could say map RANK 0,1 to GPU 0, and RANK 2 to GPU 2.